### PR TITLE
[FL-2674] Show error popup when NFC chip is not init/disconnected

### PR DESCRIPTION
**Pull request by Astrrra**
# What's new

- If rfalNfc state is not init (because the ST25R3916 chip is disconnected/broken), the NFC app will show an error popup and exit.

# Verification 

- Verify that the NFC app works as usual
- Disconnect the NFC board ribbon cable and verify that the NFC app shows an error screen

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

**Comment by skotopes** [comment]
@Astrrra  we actually have dialog service to show standard errors

**Comment by Astrrra** [comment]
@skotopes Right, didn't know about those, that's much better!

File: applications/nfc/nfc.c [hunk 1]
```diff
@@ -224,6 +224,21 @@ void nfc_show_loading_popup(void* context, bool show) {
...
+        DialogMessage* message = dialog_message_alloc();
+        dialog_message_set_text(
+            message,
+            "Error!\nNFC chip failed to start\n\n\nSend a photo of this to:\nsupport@flipperzero.one",
+            0,
+            0,
+            AlignLeft,
+            AlignTop);
+        dialog_message_show(dialogs, message);
+        return 0;
```
Comment (gornekich) [inline applications/nfc/nfc.c]: Memory leak!
After dialog_message_show() you need
1) Free message: dialog_message_free(message);
2) Close record: furi_record_close(RECORD_DIALOGS);

Comment (gornekich) [reply applications/nfc/nfc.c]: Also, I would prefer to take this to separate function
static bool nfc_is_hal_ready() {...}

and in nfc_app(){
if(!nfc_is_hal_ready()) return 0

**Review by gornekich — CHANGES_REQUESTED** [review]

File: applications/nfc/nfc.c (hunk 1 above)
Comment (Astrrra) [reply applications/nfc/nfc.c]: Fixed!

**Review by gornekich — DISMISSED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**